src/main/java/edu/hawaii/its/api/service/MemberAttributeService.java

```diff
@@ -53,7 +56,7 @@ public MemberAttributeResults getMemberAttributeResults(String currentUser, List
             throw new AccessDeniedException();
         }
         List<String> invalidUhIdentifiers = uhIdentifiers.parallelStream()
-                .filter(uhIdentifier -> !subjectService.isValidIdentifier(uhIdentifier))
+                .filter(uhIdentifier -> !subjectService.isValidIdentifier(uhIdentifier) || (grouperService instanceof OotbGrouperApiService && !ootbGroupingPropertiesService.isValidOotbUhIdentifier(uhIdentifier)))
```

Comment:
This code is the only one that added in existed service layer. This code checks whether the uhIdentifier entered by the user is valid when using the OOTB Grouper service. It performs this check when the user clicks the add or remove button with a username or uh number. The code looks up the uhIdentifier in the OOTB user list to verify its validity.        (also I checked that there's no problem on real GROUPER service too)

Reply:
I think you should put this check inside of `subjectService.isValidIdentifier()` instead.

Reply:
Originally, the implementation was done inside subjectService.isValidIdentifier(), but later the logic was moved to memberAttributeService. The Subject service, in addition to checking if a user to be added or removed is valid through memberAttribute, also retrieves grouping groups or membership groups with verifying if the current uhIdentifier is valid. Currently, in Ootb system, there is no default user upon login; instead, the current user becomes whoever logs in through cas.its.hawaii.edu, similar to a regular grouper service. However, in this situation where the current user changes with each login, already defined ootb valid users end up being filtered out, resulting in the current user being considered an invalid user. Therefore, the logic has now been separately moved to the memberAttributeService. According to the discussion with Michael today, it's likely that the login through cas.its.hawaii.edu will be removed in the ootb, and logging in as a default user will be set. If that happens, the logic might be moved back to the subject service.

Reply:
I was thinking of this:
```
public boolean isValidIdentifier(String uhIdentifier) {
    if (grouperService instanceof OotbGrouperApiService) {
        return ootbGroupingPropertiesService.isValidOotbUhIdentifier(uhIdentifier);
    }
    return isValidSubject(getSubject(uhIdentifier));
}
```
Wouldn't this avoid any of the problems you encountered?

Reply:
Oh, I just realized the logic that currently in the MemberattributeService was in the isValidSubject, not isValidIdentifier. It still has false value on the isValidOotbUhIdentifier(uhIdentifier) when uhIdentifier is current user uid, but fetching groupings and membership works well since isValidSubject is important logic for validating current user. I will change it with first code that you provided.